engine/table/src/main/java/io/deephaven/engine/table/impl/perf/UpdatePerformanceTracker.java

```diff
@@ -59,67 +60,75 @@ public class UpdatePerformanceTracker {
-    private final Table blink;
-    private final AtomicInteger entryIdCounter = new AtomicInteger(1);
-    private final Queue<WeakReference<PerformanceEntry>> entries = new LinkedBlockingDeque<>();
-
-    private UpdatePerformanceTracker(UpdateGraph updateGraph) {
-        this.updateGraph = Objects.requireNonNull(updateGraph);
+    private StreamToBlinkTableAdapter adapter;
+    private Table blink;
+    private static final AtomicInteger entryIdCounter = new AtomicInteger(1);
+
+    public UpdatePerformanceTracker() {
         this.tableLogger = EngineTableLoggers.get().updatePerformanceLogLogger();
         this.publisher = new UpdatePerformanceStreamPublisher();
-        this.adapter = new StreamToBlinkTableAdapter(
-                UpdatePerformanceStreamPublisher.definition(),
-                publisher,
-                updateGraph,
-                UpdatePerformanceTracker.class.getName());
-        this.blink = adapter.table();
     }
 
-    private void startThread() {
-        final ExecutionContext context = ExecutionContext.getContext().withUpdateGraph(updateGraph);
-        Thread driverThread = new Thread(new Driver(context), "UpdatePerformanceTracker.Driver");
-        driverThread.setDaemon(true);
-        driverThread.start();
-    }
+    public class Driver implements Runnable {
 
-    public static synchronized void start() {
-        if (started) {
-            return;
+        private final ExecutionContext context;
+        private final PerformanceEntry aggregatedSmallUpdatesEntry;
+        private final Queue<WeakReference<PerformanceEntry>> entries = new LinkedBlockingDeque<>();
+
+        private boolean started = false;
+        private boolean unitTestMode = false;
+
+        Driver(final UpdateGraph updateGraph) {
+            this.context = ExecutionContext.getContext()
+                    .withUpdateGraph(Objects.requireNonNull(updateGraph));
+            this.aggregatedSmallUpdatesEntry = new PerformanceEntry(
+                    QueryConstants.NULL_INT, QueryConstants.NULL_INT, QueryConstants.NULL_INT,
+                    "Aggregated Small Updates", null, context.getUpdateGraph().getName());
         }
-        started = true;
-        getInstance().startThread();
-    }
-
-    private class Driver implements Runnable {
 
-        private final ExecutionContext context;
+        public void start() {
+            if (started) {
+                return;
+            }
+            if (publisher == null) {
+                throw new IllegalStateException("UpdatePerformanceTracker must be started first");
+            }
+            started = true;
 
-        public Driver(ExecutionContext context) {
-            this.context = Objects.requireNonNull(context);
+            Thread driverThread = new Thread(this,
+                    "UpdatePerformanceTracker.Driver[" + context.getUpdateGraph().getName() + "]");
+            driverThread.setDaemon(true);
+            driverThread.start();
```

Comment:
We talked about moving this into the update graph, since it has to hold the lock anyway. Could be done from a terminal notification, or similar. Does not need to hold the exclusive lock.